### ISSUE TITLE
Euler beam with load over the beam span (not on the end)

### ISSUE THREAD

**Issue by nimishvj** [lululxvi/deepxde#314]
Hi Lu

I am interested in solving the Euler beam problem with a concentrated load at half length of the beam instead of the end point. I modified the Euler_beam.py code to as follows, but it gives more than 15% error for the free end displacement. can you please let me know if my implementation is correct or not? thank you.  

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import numpy as np

import deepxde as dde
from deepxde.backend import tf

def main():
    def ddy(x, y):
        return dde.grad.hessian(y, x)

    def dddy(x, y):
        return tf.gradients(ddy(x, y), x)[0] - 1.

    def pde(x, y):
        dy_xxxx = tf.gradients(dddy(x, y), x)[0]
        return dy_xxxx

    def boundary_l(x, on_boundary):
        return on_boundary and np.isclose(x[0], 0)

    def boundary_r(x, on_boundary):
        return on_boundary and np.isclose(x[0], 1)

    def boundary_cntr(x, on_boundary):
        return np.isclose(x[0], 0.5)    

    def func(x):
        return -(x ** 4) / 24 + x ** 3 / 6 - x ** 2 / 4

    geom1 = dde.geometry.Interval(0, 0.51) 
    geom2 = dde.geometry.Interval(0.5,1)
    geom3 = geom1 & geom2
    geom = geom1 | geom2 | geom3

    bc1 = dde.DirichletBC(geom, lambda x: 0, boundary_l)
    bc2 = dde.NeumannBC(geom, lambda x: 0, boundary_l)
    bc3 = dde.OperatorBC(geom, lambda x, y, _: ddy(x, y), boundary_cntr)
    bc4 = dde.OperatorBC(geom, lambda x, y, _: dddy(x, y), boundary_cntr)
    xxx=([0.25],[0.5],[0.75])
    data = dde.data.PDE(
        geom,
        pde,
        [bc1, bc2, bc3, bc4],
        num_domain=20,
        num_boundary=2,
        num_test=100,
        anchors = xxx,
    )
    layer_size = [1] + [20] * 3 + [1]
    activation = "tanh"
    initializer = "Glorot uniform"
    net = dde.maps.FNN(layer_size, activation, initializer)

    model = dde.Model(data, net)
    model.compile("adam", lr=0.001)
    losshistory, train_state = model.train(epochs=5000)
    dde.saveplot(losshistory, train_state, issave=True, isplot=False)

if __name__ == "__main__":
    main()

**Comment by lululxvi** [comment]
I am confused with your `geom`. Is it just `geom = dde.geometry.Interval(0, 1) `?

**Comment by nimishvj** [comment]
thanks, adding the BC on right end fixes the issue.
